### PR TITLE
Spellcheck Jupyter notebooks & allow running checker from other repos

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Closes https://github.com/Qiskit/documentation/issues/1741. We use the cspell library to check Jupyter notebooks:

```
❯ npm run check:spelling

> qiskit-documentation@0.1.0 check:spelling
> tsx scripts/js/commands/checkSpelling.ts

$ npx cspell --no-progress --config scripts/config/cspell/cSpell.json docs/**/*.mdx !docs/api/**/*.mdx docs/api/migration-guides/*.mdx
CSpell: Files checked: 53, Issues found: 0 in 0 files
❌ Unrecognized word in docs/guides/hello-world.ipynb: typoooooo
```

There is now an `--apis` flag, although there are currently failures.

--

Other repositories can also now run cspell with this `script`:

```
"check:spelling": "node node_modules/qiskit-documentation/dist/commands/checkSpelling.js --config node_modules/qiskit-documentation/scripts/config/cspell/cspell.json"
```

### CODE REVIEW

File: scripts/js/commands/checkSpelling.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  if (args.apis) {
+    const apiFolders = Pkg.VALID_NAMES.map((api) => `docs/api/${api}/**/*.mdx`);
+    await $`${cspellCmd} ${apiFolders}`;
+  }
+
+  await checkAllNotebooks(args.config);
+});
+
+async function checkAllNotebooks(configPath: string): Promise<void> {
```
Comment (Eric-Arellano) [inline scripts/js/commands/checkSpelling.ts:60]: Thoughts appreciated on how to make the logging more useful. Please take a look at the PR description for example output

I don't know if we want to unconditionally log that we're checking n Jupyter notebooks? Might be noisy.

Two ideas I just thought of:

* Add an emoji warning for each failure
* If it's the first failure encountered, use `\n\n` so it stands out from the logs above it.

Comment (frankharkins) [reply scripts/js/commands/checkSpelling.ts:60]: I think a "❌" emoji usually works well

File: scripts/js/commands/checkSpelling.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  const paths = await globby("docs/**/*.ipynb");
+  let allGood = true;
+  await pMap(paths, async (path) => {
+    const errors = await checkForNotebookMistakes(path, configPath);
+    if (errors.length) {
+      allGood = false;
+      const deduplicated = new Set(errors);
+      deduplicated.forEach((mistake) =>
+        console.error(`Spelling mistake in ${path}: ${mistake}`),
```
Comment (frankharkins) [inline scripts/js/commands/checkSpelling.ts:69]:
```suggestion
        console.error(`Unrecognized word in ${path}: ${mistake}`),
```
Nit to make it clearer to users that we're matching against a dictionary and they can add new words if necessary.

File: scripts/js/commands/checkSpelling.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+/**
+ * Returns any spelling mistakes in the notebook's markdown blocks, if any.
+ */
+async function checkForNotebookMistakes(
+  fp: string,
+  configPath: string,
+): Promise<string[]> {
+  // We should only check the markdown blocks, not code blocks.
+  const text = await readMarkdown(fp);
```
Comment (frankharkins) [inline scripts/js/commands/checkSpelling.ts:87]: Don't block on this, but isn't cspell able to check code too?

**Review by frankharkins — APPROVED** [review]
Nice! Very easy to read

**Pull request merged**